storages/redis/redis.go

```diff
@@ -0,0 +1,146 @@
+// Package redis contains redis storage.
+package redis
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/redis/go-redis/v9"
+	"github.com/vitaliy-ukiru/fsm-telebot"
+)
+
+type keyType string
+
+const (
+	stateKey     keyType = "state"
+	stateDataKey keyType = "data"
+)
+
+type Storage struct {
+	rds      *redis.Client
+	prefix   string
+	ttlState time.Duration
+	ttlData  time.Duration
+}
+
+type StorageSettings struct {
+	// Prefix for records in Redis.
+	// Default is "fsm".
+	prefix string
+
+	// TTL for state.
+	// Default is 0.
+	ttlState time.Duration
+
+	// TTL for state data.
+	// Default is 0.
+	ttlData time.Duration
+}
+
+// NewStorage returns new redis storage.
+func NewStorage(client *redis.Client, pref StorageSettings) fsm.Storage {
```

Comment:
make return type "*Storage" (i' ll change it in memory storage also)

Reply:
resolved in #4 https://github.com/vitaliy-ukiru/fsm-telebot/pull/4/commits/5a5f86b62b6da9b1bbe14eb23a4b54649bef6e93